modular/gater/admin_handler.go

```diff
@@ -658,12 +662,22 @@ func (g *GateModular) replicateHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (g *GateModular) checkReplicatePermission(ctx context.Context, receiveTask gfsptask.GfSpReceivePieceTask, signatureAddr string) error {
-	objectInfo, err := g.baseApp.Consensus().QueryObjectInfo(ctx, receiveTask.ObjectInfo.BucketName, receiveTask.ObjectInfo.ObjectName)
+	var (
+		objectInfo *storagetypes.ObjectInfo
+		err        error
+	)
+	for retry := 0; retry < checkPermissionRetry; retry++ {
+		objectInfo, err = g.baseApp.Consensus().QueryObjectInfo(ctx, receiveTask.ObjectInfo.BucketName, receiveTask.ObjectInfo.ObjectName)
```

Comment:
if we can't fetch objectInfo from chain, can we just retrieve it from SP metadata service?

Reply:
It takes longer for meta  svc to have objects data stored.